compiler/version_check.go

```diff
@@ -1,9 +1,25 @@
-// +build go1.12
-// +build !go1.13
-
 package compiler
 
-const ___GOPHERJS_REQUIRES_GO_VERSION_1_12___ = true
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+)
 
 // Version is the GopherJS compiler version string.
 const Version = "1.12-2"
+
+// CheckGoVersion checks the version of the Go distribution
+// at goroot, and reports an error if it's not compatible
+// with this version of the GopherJS compiler.
+func CheckGoVersion(goroot string) error {
+	v, err := ioutil.ReadFile(filepath.Join(goroot, "VERSION"))
+	if err != nil {
+		return fmt.Errorf("GopherJS %s requires a Go 1.12.x distribution, but failed to read its VERSION file: %v", Version, err)
+	}
+	if !bytes.Equal(v, []byte("go1.12")) && !bytes.HasPrefix(v, []byte("go1.12.")) {
```

Comment:
I'd want `strings.TrimSpace` for `v` just in case.

This check would fail with e.g., `go1.12rc1`.

Reply:
Ok, I'll make it just check for a "go1.12" prefix. The code will change before we need to worry about it matching Go 1.120 falsely.

Reply:
`/^go1.12([.a-z].+)?/` maybe?

Reply:
Done. I think the prefix check should be sufficient for now.

Reply:
As this string comes from a file, I'd want `strings.TrimSpace`.

Reply:
On the second thought, `HasPrefix` would be enough to ignore the tailing new line char.

Reply:
Thanks for suggesting `strings.TrimSpace`. I've considered it, but I don't think it's needed. In a normal Go installation, there are never leading spaces.